lib/matplotlib/artist.py

```diff
@@ -898,6 +900,89 @@ def set_zorder(self, level):
         self.pchanged()
         self.stale = True
 
+    def get_top_margin(self):
```

Comment:
:+1: on the single code path for set/update/setp.  I like that idea a lot.

I have a lot less objection doing both property and getter/setter.  In fact, we can probably implement it fairly easily using the "old" pre-decorator property style.  Yes, it will look archaic, but it's the best way to reduce code duplication IMHO.

```
def get_FOO(self):
   ...

def set_FOO(self, val):
   ...

FOO = property(get_FOO, set_FOO)
```